lib/trino-filesystem-s3/src/test/java/io/trino/filesystem/s3/AbstractTestS3FileSystem.java

```diff
@@ -134,19 +146,30 @@ void testFileWithTrailingWhitespaceAgainstNativeClient()
         try (S3Client s3Client = createS3Client()) {
             String key = "foo/bar with whitespace ";
             byte[] contents = "abc foo bar".getBytes(UTF_8);
-            s3Client.putObject(
-                    request -> request.bucket(bucket()).key(key),
-                    RequestBody.fromBytes(contents.clone()));
+
+            PutObjectRequest putObjectRequest = PutObjectRequest.builder()
+                    .bucket(bucket())
+                    .key(key)
+                    .applyMutation(builder -> {
+                        if (useServerSideEncryptionWithCustomerKey()) {
+                            builder.sseCustomerAlgorithm(AES256.toString());
+                            builder.sseCustomerKey(encoded(randomEncryptionKey));
+                            builder.sseCustomerKeyMD5(md5Checksum(randomEncryptionKey));
+                        }
+                    })
+                    .build();
+
+            s3Client.putObject(putObjectRequest, RequestBody.fromBytes(contents.clone()));
             try {
                 // Verify listing
-                List<FileEntry> listing = toList(fileSystem.listFiles(getRootLocation().appendPath("foo")));
+                List<FileEntry> listing = toList(getFileSystem().listFiles(getRootLocation().appendPath("foo")));
```

Comment:
Move this refactoring to a separate commit

Reply:
Extracted